xmldiff/diff.py

```diff
@@ -162,23 +162,31 @@ def node_ratio(self, left, right):
             # One is a comment the other is not:
             return 0
 
+        match = None
         for attr in self.uniqueattrs:
-            if not isinstance(attr, str):
-                # If it's actually a sequence of (tag, attr), the tags must
-                # match first.
-                tag, attr = attr
-                if tag != left.tag or tag != right.tag:
-                    continue
-            if attr in left.attrib or attr in right.attrib:
-                # One of the nodes have a unique attribute, we check only that.
+            if isinstance(attr, dict):
+                for tag, attr_list in attr.items():
```

Comment:
We could shortcut the whole test here, if we check that the tags are the same before the loop.
Then you also need only `if tag != left.tag` below, as the tags now are known to be the same.